modules/currency-l1/src/main/scala/org/tessellation/currency/l1/CurrencyL1App.scala

```diff
@@ -86,6 +88,10 @@ abstract class CurrencyL1App(
           p2pClient,
           cfg
         )
+
+      // TODO is this the correct starting value?
+      isAlignedR <- Ref.of[IO, Boolean](true).asResource
```

Comment:
I know it's a draft PR but two things came to my mind. Are we sure we want to keep such a thing at the very top level (currency L1 app)? IMO there should be some job/worker checking the alignment and reacting in some way in case of misalignment (see redownload logic that is already implemented on DAG L1, cc: @TheMMaciek). This looks like mixed responsibilities

Reply:
I'm not sure we need this top level check here for the L1, since the redownload is capable of mutating the state. In the L0 rollback program, where there that process currently runs as a main method invocation -- then to re-use that you might need some top level invocation if you're unable to mutate the currency L0 state otherwise. But I don't think that's necessary in L1 since you can mutate there following the pattern of the redownload as marcin is saying.

Reply:
It would be better to use a top level semaphore if needed, instead of the boolean Ref, but I'd not recommend that.